backend/manager/modules/common/src/main/java/org/ovirt/engine/core/common/businessentities/VdsCpuUnit.java

```diff
@@ -3,12 +3,21 @@
 import java.io.Serializable;
 import java.util.Objects;
 
-public class VdsCpuUnit implements Serializable {
+import org.ovirt.engine.core.compat.Guid;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+
+
+public class VdsCpuUnit implements Comparable<VdsCpuUnit>, Serializable, Cloneable {
 
     private static final long serialVersionUID = -397254497953366129L;
     private int core;
     private int socket;
     private int cpu;
+    @JsonIgnore
+    private Guid vmId;
```

Comment:
I will have to change it to a list.
Unfortunately, after today meeting. I thought of the case of 2 VMs(or more) using the same pCPU. 
Once the VM will go down we will clear the pCPU, although it is under usage of another VM.
This is a missing gap to support the manual combining with dedicated.

Reply:
yeah, it is more tricky with manual pinning because of that mapping of a CPU to multiple VMs
not sure about trying to maintain this correlation of a pCPUs to VMs - do we really that for scheduling?

Reply:
That will be really helpful for API. You can know easily which VM took which pCPUs.

Reply:
yes, it can be useful for the API or the UI but those clients don't have to return an accurate information (the information has to be consistent but not necessarily the latest) - if having this information on this structure complicates the backup logic then I wouldn't insist on having it here, we can also query the database to retrieve this information for clients

Reply:
I agree with Lucia.

Reply:
> What do you meant that it is not necessary to return an accurate information? That is the purpose of the dialog for displaying of the currently pinned CPUs - current and accurate information.

ideally yes, but practically most of the dialogs we have may display inaccurate data - imagine you open edit-vm on one tab and then modify the VM elsewhere, that change won't be reflected on the first tab, right? here it is even worse - imagine you open the pinning dialog and a vm with cpu pinning starts later, then unless you keep reloading the data upon change or frequent enough, the dialog won't show the up-to-date (what I called "accurate" before) data.. we shouldn't rely on the dialog containing the up-to-date information because we cannot guarantee that

> I am for changing this to a list as I suppose it will not be a big problem on the backend. A nice new methods pinVm(vmId, cpuPinningPolicy) and unpinVm(vmId), isPinned(vmId) should work.
> 
> We can even put some constraints to the add method to check that we're not adding vm with manual policy to a cpu already dedicated etc.

long time ago we asked ourselves why having this information cached in VdsManager and the idea was to prevent the scheduler from retrieving the data from the database and constructing this view again and again for each run/migrate command. sure, you can enhance it with more methods and concurrent data structure - that's possible but the downside is code complexity and the amount of data we store in-memory. in my view, the less places the interact with this data structure (ideally, only the scheduler and ProcessDownVm) the better because we'll have less synchronization issues

Reply:
Well, all you've written would be true, only the data structure would be different. If you do not like the naming, forget about the suggested method names, it could be named differently, it was just a suggestion (but you'd need to have methods for adding/removing/checking the vm id from/to list)